designs/logging.rst

```diff
@@ -0,0 +1,170 @@
+Logging Design (Refactor)
+-------------------------
+
+Logging is how the server communicates to developers and administrators about the state
+of the service, and how operations are performing and what they are doing. It's important
+this is clear in how it communicates. Today (2020-05-12) the log has been written with
+development in mind, and has a structure that has as a result, become hard to parse and
+understand. This has motivated a rewrite of logging to improve how the servers state
+and errors are communicated to users.
+
+Use Cases
+---------
+
+* Developer Bug Reports
+
+A developer should be able to see the internal state of the server, and why any decision
+or logic path was taken, so that errors or logic can be analysed post-incident. The
+information in the log and the source code should be enough to resolve any issues, as we
+may not have LLDB access to any consumers site, or any effective reproducer.
+
+* Security Audits
+
+We must be able to see why any security decision was made, such as credential validation,
+access control application, or group/claim issuing to a session. This should be connected
+to the IP and other identifiers of the caller.
+
+* Error Analysis
+
+For an administrator, they must be able to determine why an operation is failing in detail
+so they can advise on how a consumer or user could change their behaviour to improve the
+situation (beyond the error messages we return).
+
+* Performance
+
+Administrators and Developers should be able to analyse fine grained information about the
+performance of any operation, and make informed decisions about tuning (such as caches or
+or threads), and developers should be able to identify code paths that are under pressure
+and could be targets for improvement.
+
+* Containers
+
+Logs should be access via an API, and support some querying or extraction that can be
```

Comment:
I think that usually working with containers is enough to write in stdout/stderr. Admins must ensure to take logs and put in a place to be reviewed and accessible (via API or whatever).

Reply:
That's a good point. I think I will make it stdout/stderr only to start with just because it will work with systemd as well that way. So I'll updated this part, and then merge. Thanks!